here-naksha-app-service/src/main/java/com/here/naksha/app/service/http/tasks/ReadFeatureApiTask.java

```diff
@@ -214,4 +217,37 @@ protected void init() {}
     // transform Result to Http FeatureCollection response, restricted by given feature limit
     return transformReadResultToXyzCollectionResponse(result, XyzFeature.class, limit);
   }
+
+  private @NotNull XyzResponse executeIterate() {
+    // Parse and validate Path parameters
+    final String spaceId = ApiParams.extractMandatoryPathParam(routingContext, SPACE_ID);
+
+    // Parse and validate Query parameters
+    final QueryParameterList queryParams = (routingContext.request().query() != null)
+        ? new QueryParameterList(routingContext.request().query())
+        : null;
+    // Note : subsequent steps need to support queryParams being null
+
+    // extract limit parameter
+    long offset = 0;
+    long limit = ApiParams.extractQueryParamAsLong(queryParams, LIMIT, false, DEF_FEATURE_LIMIT);
+    // extract handle parameter
+    IterateHandle handle = ApiParams.extractQueryParamAsIterateHandle(queryParams, HANDLE);
+    // create new "handle" if not already provided, or overwrite parameters based on "handle"
+    if (handle == null) {
+      handle = new IterateHandle().withOffset(offset).withLimit(limit);
+    }
+    offset = handle.getOffset();
```

Comment:
The lifecycle of `offset` variable got me confused a little:
1. We set `offset` to 0 in L:232
2. If handle is null, we create handle with `offset` (which internally is a simple setter)
3. We override `offset` with what's has been defined as handle's property.

Can't we initialize offset in line 240 without even mentioning it before? `IterateHandle` could probably have default `offset` value set to 0

Reply:
2nd step is redundant (as in other comment).
However setting offset at 240 with what is there in handle value is important and it can be different if handle is not null (that means, when handle is supplied as part of API input. In that case, offset will have some value other than default i.e. 0).